content/courses/advanced-ada/parts/data_types/arrays.rst

```diff
@@ -52,21 +73,59 @@ also highlights the fact that the bounds are fixed once an object is declared:
 In the :ada:`Show_Measurements` procedure above, once we declare :ada:`M`, its
 bounds are fixed for the whole lifetime of :ada:`M`. We cannot *add* another
 component to this array. In other words, :ada:`M` will have 10 components for
-its whole lifetime.
+its whole lifetime:
 
-.. admonition:: In the Ada Reference Manual
+.. code-block:: ada
 
-    - :arm22:`3.6 Array Types <3-6>`
+    M : Measurements (1 .. 10);
+    --                ^^^^^^^
+    --  Bounds cannot be changed!
+
+
+.. _Adv_Ada_Constrained_Array_Type:
+
+Constrained array types
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Note that we could declare constrained array types. Let's rework the previous
+example:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Data_Types.Arrays.Array_Constraints.Constrained_Array_Type
+
+    package Measurement_Defs is
+
+       type Measurements is
+         array (1 .. 10) of Float;
+       --       ^ Bounds are of known and fixed.
+
+    end Measurement_Defs;
+
+    with Ada.Text_IO;      use Ada.Text_IO;
+
+    with Measurement_Defs; use Measurement_Defs;
+
+    procedure Show_Measurements is
+       M : Measurements;
+       --              ^ We cannot change the
+       --                bounds here!
+    begin
+       Put_Line ("First index: " & M'First'Image);
+       Put_Line ("Last index:  " & M'Last'Image);
+    end Show_Measurements;
+
+In this case, the bounds of the :ada:`Measurements` type are fixed. Now, we
+cannot specify the bounds (or change them) in the declaration of the :ada:`M`
+array, as they have already been defined in the type declaration.
 
 
 Unconstrained Arrays vs. Vectors
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 If you need, however, the flexibility of increasing the length of an array, you
 could use vectors instead. This is how we could rewrite the previous example
```

Comment:
Maybe "could use the language-defined Vector type instead."

Reply:
Yes, that sounds better. Thanks for the suggestion!